src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbolFromDesignation.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents expression variables declared in a global statement.
+    /// </summary>
+    internal class SourceMemberFieldSymbolFromDesignation : SourceMemberFieldSymbol
```

Comment:
>SourceMemberFieldSymbolFromDesignation [](start = 19, length = 38)

This is poorly named. Specifically, the only thing that makes it specific to a designation is that it has constructors that receive a designation and store particular parts of it. Forseeing that me might want to handle other kinds of locals lifted to fields (e.g. pattern variables), you might have thought that this would have to be renamed to describe what it is rather than how it is created. On the other hand, you can just as well leave that as additional work for the next person to do.